code/game/objects/items/devices/traitordevices.dm

```diff
@@ -143,4 +143,62 @@ effective or pretty fucking useless.
 
 	attack_self(usr)
 	add_fingerprint(usr)
-	return
\ No newline at end of file
+	return
+
+/obj/item/device/shadowcloak
+	name = "cloaker belt"
+	desc = "Makes you invisible for short periods of time. Recharges in darkness."
+	icon = 'icons/obj/clothing/belts.dmi'
+	icon_state = "utilitybelt"
+	item_state = "utility"
+	slot_flags = SLOT_BELT
+	attack_verb = list("whipped", "lashed", "disciplined")
+
+	var/mob/living/carbon/human/user = null
+	var/charge = 300
+	var/max_charge = 300
+	var/on = 0
+	action_button_name = "Toggle Cloaker"
+
+/obj/item/device/shadowcloak/ui_action_click()
+	if(usr.get_item_by_slot(slot_belt) == src)
+		if(!on)
+			Activate(usr)
+		else
+			Deactivate()
+	return
+
+/obj/item/device/shadowcloak/proc/Activate(mob/living/carbon/human/user)
+	if(!user)
+		return
+	user << "<span class='notice'>You activate [src].</span>"
+	src.user = user
+	SSobj.processing |= src
+	on = 1
+	return
+
+/obj/item/device/shadowcloak/proc/Deactivate()
+	user << "<span class='notice'>You deactivate [src].</span>"
+	SSobj.processing.Remove(src)
+	if(user)
+		user.alpha = initial(user.alpha)

```

Comment:
This like scares me. It could cause trouble if we have anything that changes alpha and expects it to stay changed. Although, the only case I can think off the top of my head is invisimin, and that isn't that big of a deal. Though I would rather have this be stored in a `var/old_alpha` on `Activate()` either way.